string serialized_head
---
bool success
